Mage/src/main/java/mage/abilities/effects/common/ExileCardsFromTopOfLibraryTargetEffect.java

```diff
@@ -1,39 +1,37 @@
 package mage.abilities.effects.common;
 
 import mage.abilities.Ability;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.dynamicvalue.common.StaticValue;
 import mage.abilities.effects.OneShotEffect;
 import mage.cards.Cards;
 import mage.cards.CardsImpl;
 import mage.constants.Outcome;
 import mage.constants.Zone;
 import mage.game.Game;
 import mage.players.Player;
-import mage.util.CardUtil;
 
 /**
- * @author LevelX2
+ * @author LevelX2, Susucr
  */
 public class ExileCardsFromTopOfLibraryTargetEffect extends OneShotEffect {
 
-    int amount;
-    String targetName;
+    private final DynamicValue amount;
 
-    public ExileCardsFromTopOfLibraryTargetEffect(int amount) {
-        this(amount, null);
+    public ExileCardsFromTopOfLibraryTargetEffect(int amount, String targetName) {
+        this(StaticValue.get(amount), targetName);
     }
 
-    public ExileCardsFromTopOfLibraryTargetEffect(int amount, String targetName) {
+    public ExileCardsFromTopOfLibraryTargetEffect(DynamicValue amount, String targetName) {
         super(Outcome.Exile);
-        this.amount = amount;
+        this.amount = amount.copy();
         this.staticText = (targetName == null ? "that player" : targetName) + " exiles the top "
```

Comment:
rather than null and ternary, just set "that player" in the constructor without targetName parameter

Reply:
Actually no usage with null in the codebase (only 5 cards including Ashiok do use the effect). So can remove the ternary.